sqlglot/parser.py

```diff
@@ -1551,10 +1574,55 @@ def extend_props(temp_props: t.Optional[exp.Properties]) -> None:
             begin=begin,
             end=end,
             clone=clone,
-            start=seq_start,
-            increment=seq_increment,
         )
 
+    def _parse_create_sequence(self, replace, properties) -> exp.Create | exp.Command:
+        exists = self._parse_exists(not_=True)
+        this = self._parse_table(schema=True)
+
+        seq = exp.Create(
+            this=this,
+            exists=exists,
+            replace=replace,
+            kind="SEQUENCE",
+            properties=properties,
+        )
```

Comment:
This doesn't look correct, we shouldn't be duplicating logic like that in here. Probably better to keep the branch in `_parse_create` and factor the rest of the logic (options, properties etc) out into a separate method.

Reply:
With `CREATE SEQUENCE` being it's own AST node, does this comment still hold? Bear in mind that we might have to parse some modifiers (e.g. `TEMPORARY`, `OR REPLACE`) before we get to the `SEQUENCE` token which would dispatch `_parse_create_sequence()`, and if we allow the control flow to continue to `_parse_create` we may have to multiplex between `exp.Create` and `exp.CreateSequence` parts. Also, since dialects have some differences between them, I thought that by parsing under a `while` would require that the expression is created beforehand, so I can utilize `exp.set(...)`.

Reply:
The suggestion is to parse this into `exp.Create` like all other DDLs. Let's sync on Slack for how we can clean this up.